gms/gossiper.cc

```diff
@@ -1841,11 +1841,13 @@ void gossiper::examine_gossiper(utils::chunked_vector<gossip_digest>& g_digest_l
 }
 
 future<> gossiper::start_gossiping(int generation_nbr, std::map<application_state, versioned_value> preload_local_states, gms::advertise_myself advertise) {
-    return container().invoke_on_all([advertise] (gossiper& g) {
-        if (!advertise) {
-            g._advertise_myself = false;
-        }
-    }).then([this, generation_nbr, preload_local_states] () mutable {
+    return seastar::async([this, g = this->shared_from_this(), generation_nbr, preload_local_states, advertise] () mutable {
+        container().invoke_on_all([advertise] (gossiper& g) {
+            if (!advertise) {
+                g._advertise_myself = false;
+            }
+        }).get();
```

Comment:
Usually in async code, then(...) is preferred to get(). It probably produces same result here but it would have been nice to stick to it to reduce amount of changed code.

Reply:
I tried but could not figure out how to make it work properly, also I'm not sure sure then is guaranteed to execute on the same shard

Reply:
Also, I followed the structure of do_stop_gossiping, so it is consistent with the rest of the code.
The upstream patch is much nicer in this regard: https://github.com/scylladb/scylla/pull/10548/files but the refactoring that brings co_await is not in 4.6.3 (the upstream patch is on master)